lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2089,26 +2107,16 @@ void CodeGenModule::EmitGlobalDefinition(GlobalDecl GD, llvm::GlobalValue *GV) {
   if (LangOpts.CPlusPlusAMP && !CodeGenOpts.AMPCPU) {
     if (CodeGenOpts.AMPIsDevice) {
       // If -famp-is-device switch is on, we are in GPU build path.
-      if (!isWhiteListForHCC(*this, GD))
-        return;
-    } else {
-      // In host path:
```

Comment:
@AlexVlx wouldn't these removed lines cause functions/variables with `restrict(amp)` be emitted in host path?